src/main/groovy/io/seqera/wave/service/conda/AbstractCondaFetcher.groovy

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Wave, containers provisioning service
+ *  Copyright (c) 2023-2024, Seqera Labs
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.seqera.wave.service.conda
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+import io.seqera.wave.configuration.BuildConfig
+import io.seqera.wave.service.persistence.CondaPackageRecord
+import io.seqera.wave.service.persistence.PersistenceService
+import jakarta.inject.Inject
+/**
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+abstract class AbstractCondaFetcher implements CondaFetcherService{
+
+    @Inject
+    private BuildConfig buildConfig
+
+    @Inject
+    private PersistenceService persistenceService
+
+    private List<String> channels = ['seqera']
+
+    protected List<String> fetchCommand(String channel, String target) {
+        List.of('bash','-c', "conda search -q -c $channel | head > $target".toString())
```

Comment:
@pditommaso why head is used here?
do we only need first ten packages or all of them?
 and when I am using head its failing with `write /dev/stdout: broken pipe`

Reply:
I should have mentioned it. That's only for the sake of making testing faster

Reply:
ok thank you for the clarification